MPChartLib/src/main/java/com/github/mikephil/charting/data/BaseDataSet.java

```diff
@@ -226,6 +234,16 @@ public void setColor(int color, int alpha) {
         setColor(Color.argb(alpha, Color.red(color), Color.green(color), Color.blue(color)));
     }
 
+    /**
+     * Sets the start and end color for gradient color, ONLY color that should be used for this DataSet.
+     *
+     * @param startColor
+     * @param endColor
+     */
+    public void setGradientColor(int startColor, int endColor) {
+        mGradientColor = new GradientColor(startColor, endColor);
+    }
```

Comment:
**Suggestion**: The introduction of gradient coloring to datasets is a significant enhancement. Ensure that this feature is compatible with all chart types in the library and does not interfere with existing coloring options. <br> **Code Suggestion**: 
 ```
 # Test gradient coloring extensively with various datasets and chart types to ensure broad compatibility and no regression in existing functionality.
 ```